ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -317,6 +318,15 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
       if ((runtimeVersion != null || ci.initiatorVersion != null) && !runtimeVersion.equals(ci.initiatorVersion)) {
         LOG.warn("Worker and Initiator versions do not match. Worker: v{}, Initiator: v{}", runtimeVersion, ci.initiatorVersion);
       }
+
+      if (StringUtils.isBlank(getPoolName()) && StringUtils.isNotBlank(ci.poolName)) {
+        LOG.warn("A timed out copmaction pool entry ({}) is picked up by one of the default compaction pool workers.", ci);
+      }
+      if (StringUtils.isNotBlank(getPoolName()) && StringUtils.isNotBlank(ci.poolName) && !getPoolName().equals(ci.poolName)) {
```

Comment:
id it a valid usecase?

Reply:
This normally should not happen at all, because the query filters the items by pool name. However, I wanted to cover this case as well. If a labeled (not the default) pool somehow gets a request assigned to another or the default pool, I think it should not be processed. 
- Simply skipping it could be problematic if the item gets returned by `findNextCompact()` again and again. In this case this item will stuck in `initiated` state, and if there's only one worker assigned to the pool, it will even stuck the entire queue processing for that pool. 
- Marking it as failed with a proper error message seemed to be more convenient, which won't cause processing anomalies and is easier to track down later.

Reply:
that is a non-realistic situation, but still, this logic would confuse me, why do I need to abandon a valid compaction request, it might be picked up by another worker pool or even default one after the timeout.

Reply:
I change the code that it logs a warning that somehow an entry belonging to another pool is ended up there. Regarding the min size, I'm convinced that it's a valid use case: On some of the worker nodes you may want to process only labeled queue items, and you dedicate one or more nodes for default items. However, allowing 0 means that it can be configured in a way that default items are not processed at all. But checking this would required a cluster level validation.

Reply:
ok, in that case, it would just lie in a queue until the user enables the default pool. monitoring should highlight the issue